homeassistant/components/fastdotcom/services.py

```diff
@@ -0,0 +1,48 @@
+"""Services for the Fastdotcom integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import issue_registry as ir
+
+from .const import DOMAIN, SERVICE_NAME
+from .coordinator import FastdotcomDataUpdateCoordindator
+
+
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Set up the service for the Fastdotcom integration."""
+
+    @callback
+    def collect_coordinator() -> FastdotcomDataUpdateCoordindator:
+        """Collect the coordinator Fastdotcom."""
+        config_entries = hass.config_entries.async_entries(DOMAIN)
+        for config_entry in config_entries:
+            if config_entry.state != ConfigEntryState.LOADED:
+                raise HomeAssistantError(f"{config_entry.title} is not loaded")
+            coordinator: FastdotcomDataUpdateCoordindator = hass.data[DOMAIN][
+                config_entry.entry_id
+            ]
+            break
+        return coordinator
```

Comment:
We need to handle if there are no config entries for the domain, eg if the config entry is removed and Home Assistant hasn't restarted yet. Raise `HomeAssistantError` that there are no loaded config entries for the integration.